src/unix_socket.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::Result;
+use reqwest::blocking::{Request, Response};
+use reqwest::header::{HeaderValue, HOST};
+use std::path::PathBuf;
+use std::time::Instant;
+
+use crate::middleware::{Context, Middleware, ResponseMeta};
+
+pub struct UnixSocket {
+    rt: tokio::runtime::Runtime,
+    socket_path: PathBuf,
+}
+
+impl UnixSocket {
+    pub fn new(socket_path: PathBuf) -> Self {
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        Self { rt, socket_path }
+    }
+
+    pub fn execute(&self, request: Request) -> Result<Response> {
+        self.rt.block_on(async {
+            // TODO: Support named pipes by replacing tokio::net::UnixStream::connect(..) with:
+            //
+            // use std::time::Duration;
+            // use tokio::net::windows::named_pipe;
+            // use windows_sys::Win32::Foundation::ERROR_PIPE_BUSY;
+            //
+            // let stream = loop {
+            //     match named_pipe::ClientOptions::new().open(r"\\.\pipe\docker_engine") {
+            //         Ok(client) => break client,
+            //         Err(e) if e.raw_os_error() == Some(ERROR_PIPE_BUSY as i32) => (),
+            //         Err(e) => return Err(e)?,
+            //     }
+            //
+            //     tokio::time::sleep(Duration::from_millis(50)).await;
+            // };
+            let stream = tokio::net::UnixStream::connect(&self.socket_path).await?;
+
+            let (mut sender, conn) = hyper::client::conn::http1::Builder::new()
+                .title_case_headers(true)
+                .handshake(hyper_util::rt::TokioIo::new(stream))
+                .await?;
+
+            tokio::task::spawn(async move {
+                if let Err(err) = conn.await {
+                    log::error!("Connection failed: {:?}", err);
+                }
+            });
+
+            // TODO: don't ignore value from --timeout option
```

Comment:
Regardless of how this is handled, I need to ensure the correct exit code is used for timeout errors